### DISCUSSION TITLE
How can I unit test to catch memory leak?

### DISCUSSION THREAD

**Discussion by the-yuyut** [Quick/Quick#1162] [Q&A]
In XCTest, I can have [addTeardownBlock](https://developer.apple.com/documentation/xctest/xctestcase/2887226-addteardownblock) do detect weak copy is nil or not
```swift
let sut = makeSUT(...)

func makeSUT(...) -> SUT {
   let sut = ... 
   addTearDownBlock { [weak sut] in 
       XCTAssertNil(sut)
   }
   return sut
}
```

But I have no idea to to do that by Quick, I have done some search, and I found #1132  and its related discussion, but still have no clue.

Can this be done by `aroundEach`?

**Comment by younata** [comment]
Hey @ytyubox! Thanks for reaching out!

Yes, `aroundEach` should be able to do this for you, I'd write that test like so:

```swift
var sut: Subject?
aroundEach { example in
    autoreleasepool {
        sut = makeSut(...)
        example()
    }
    expect(sut).to(beNil())
}
```

Which should do what you want, but I didn't verify that code myself.

You can also still use the XCTest-based methodology with `addTearDownBlock` inside of a test, like so:

```swift
it("deallocates correctly") {
    let sut = makeSUT(...)
    addTearDownBlock { [weak sut] in expect(sut).to(beNil()) }
    return sut
}
```

Hope that helps!
